engine/src/main/java/org/hibernate/validator/internal/util/logging/Log.java

```diff
@@ -546,8 +546,8 @@ UnexpectedTypeException getNoValidatorFoundForTypeException(String constraintTyp
 	ValidationException getValidateOnExecutionOnOverriddenOrInterfaceMethodException(Method m);
 
 	@Message(id = 167,
-			value = "A given constraint definition can only be overridden in one mapping file. %1$s is overridden in multiple files")
-	ValidationException getOverridingConstraintDefinitionsInMultipleMappingFilesException(String constraintClass);
+			value = "A given constraint definition can only be overridden once (either through a XML file or the programmatic API). %1$s is overridden in multiple XML files and/or API calls")
```

Comment:
I think that deviates a bit from the behavior of the DSL in other places. There, constraint mappings given via the DSL would be _added_ to the ones given via annotations and/or XML, unless `ignoreAnnotations()` is called (the name seems sub-par, but as per the ref guide it applies to XML, too).

So should we similarly _add_ programmatically given definitions, unless `includeExistingValidators()` is called?